lib/shared/pages/change_max_amount_bottom_sheet.dart

```diff
@@ -0,0 +1,218 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+import 'package:givt_app/app/injection/injection.dart';
+import 'package:givt_app/features/account_details/bloc/personal_info_edit_bloc.dart';
+import 'package:givt_app/features/auth/cubit/auth_cubit.dart';
+import 'package:givt_app/l10n/l10n.dart';
+import 'package:givt_app/shared/widgets/widgets.dart';
+import 'package:givt_app/utils/app_theme.dart';
+import 'package:go_router/go_router.dart';
+
+class ChangeMaxAmountBottomSheet extends StatelessWidget {
+  const ChangeMaxAmountBottomSheet({
+    required this.icon,
+    required this.maxAmount,
+    super.key,
+  });
+
+  final int maxAmount;
+  final IconData icon;
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (context) => PersonalInfoEditBloc(
+        authRepositoy: getIt(),
+        loggedInUserExt: context.read<AuthCubit>().state.user,
+      ),
+      child: _ChangeMaxAmountBottomSheetView(
+        icon: icon,
+        maxAmount: maxAmount,
+      ),
+    );
+  }
+}
+
+class _ChangeMaxAmountBottomSheetView extends StatefulWidget {
+  const _ChangeMaxAmountBottomSheetView({
+    required this.icon,
+    required this.maxAmount,
+  });
+
+  final int maxAmount;
+  final IconData icon;
+
+  @override
+  State<_ChangeMaxAmountBottomSheetView> createState() =>
+      _ChangeMaxAmountBottomSheetViewState();
+}
+
+class _ChangeMaxAmountBottomSheetViewState
+    extends State<_ChangeMaxAmountBottomSheetView> {
+  late TextEditingController amount;
+  bool get isEnabled {
+    final enteredAmount = int.parse(amount.text);
+    if (enteredAmount == 0) {
+      amount.selection = TextSelection(
+        baseOffset: 0,
+        extentOffset: amount.text.length,
+      );
+      return false;
+    }
+    if (enteredAmount == widget.maxAmount) {
+      return false;
+    }
+    return true;
+  }
+
+  void increaseAmount() {
+    if (int.parse(amount.text) == 99999) {
+      return;
+    }
+    setState(() {
+      amount.text = (int.parse(amount.text) + 5).toString();
```

Comment:
same here with 5. not sure what is it? 
maybe const step = 5; ?

Reply:
Yeah, agreed